src/Domain/Creators/TimeDependentOptions/ExpansionMap.cpp

```diff
@@ -4,87 +4,206 @@
+      result =
+          volume_fot.at(name)->create_at_time(initial_time, expiration_time);
+    } else {
+      ASSERT(AllowSettleFoTs,
+             "Expansion function of time in the volume file is a "
+             "SettleToConstant, but SettleToConstant functions of time aren't "
+             "allowed.");
+      result = volume_fot.at(name)->get_clone();
+    }
+  } else if (std::holds_alternative<ExpansionMapOptions<AllowSettleFoTs>>(
+                 expansion_map_options)) {
+    const auto& hard_coded_options =
+        std::get<ExpansionMapOptions<AllowSettleFoTs>>(expansion_map_options);
+
+    if (hard_coded_options.asymptotic_velocity_outer_boundary.has_value()) {
+      result =
+          std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<2>>(
+              initial_time, hard_coded_options.initial_values, expiration_time);
+    } else {
+      ASSERT(hard_coded_options.decay_timescale.has_value(),
+             "To construct an ExpansionMap SettleToConstant function of time, "
+             "a decay timescale must be supplied.");
+      result = std::make_unique<domain::FunctionsOfTime::SettleToConstant>(
+          hard_coded_options.initial_values, initial_time,
+          hard_coded_options.decay_timescale.value());
+    }
+  } else {
+    ERROR("Unknown ExpansionMap.");
+  }
+
+  return result;
+}
+
+template <bool AllowSettleFoTs>
+std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>
+get_expansion_outer_boundary(
+    const std::variant<ExpansionMapOptions<AllowSettleFoTs>, FromVolumeFile>&
+        expansion_map_options,
+    const double initial_time) {
+  const std::string name = "ExpansionOuterBoundary";
+  std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime> result{};
+
+  if (std::holds_alternative<FromVolumeFile>(expansion_map_options)) {
+    const auto& from_vol_file = std::get<FromVolumeFile>(expansion_map_options);
+    const auto volume_fot =
+        from_vol_file.retrieve_function_of_time({name}, initial_time);
+
+    // It must be either a FixedSpeedCubic or a SettleToConstant
+    const auto* cubic_volume_fot =
+        dynamic_cast<domain::FunctionsOfTime::FixedSpeedCubic*>(
+            volume_fot.at(name).get());
+    const auto* settle_volume_fot =
+        dynamic_cast<domain::FunctionsOfTime::SettleToConstant*>(
+            volume_fot.at(name).get());
+
+    if (UNLIKELY(cubic_volume_fot == nullptr and
+                 settle_volume_fot == nullptr)) {
+      ERROR_NO_TRACE(
+          "ExpansionOuterBoundary function of time read from volume data is "
+          "not a FixedSpeedCubic or a SettleToConstant. Cannot use it to "
+          "initialize the expansion map.");
+    }
+
+    ASSERT(cubic_volume_fot != nullptr or AllowSettleFoTs,
+           "ExpansionOuterBoundary function of time in the volume file is a "
+           "SettleToConstant, but SettleToConstant functions of time aren't "
+           "allowed.");
+
+    result = volume_fot.at(name)->get_clone();
```

Comment:
(optional) I think you can always just return `create_at_time()` because it defaults to `get_clone()`, then you don't have to decide if the FoT actually does something special in `create_at_time`.